packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2599,6 +2602,7 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
       removeBottom: removeBottomPadding,
     );
     if (removeBottomInset) {
+      onRemoveBottomInsets?.call(data.viewInsets);
```

Comment:
Calling a callback from a build function is an odd side effect of build. Also, whether the bottomInsets are removed is user-controlled via the `resizeToAvoidBottomInset` property. Why do you need a callback for something that's fully in your control?

Reply:
Because resizeToAvoidBottomInset is very useful, and sometimes you want to use the bottom inset to control something that is outside the scaffold, this just keeps track of the removed bottom inset